test/javascripts/controllers/assign-user-test.js

```diff
@@ -2,37 +2,42 @@ import EmberObject from "@ember/object";
 import pretender from "discourse/tests/helpers/create-pretender";
 import { discourseModule } from "discourse/tests/helpers/qunit-helpers";
 import { test } from "qunit";
-import { Promise } from "rsvp";
 
 discourseModule("Unit | Controller | assign-user", function () {
-  test("doesn't set suggestions and fails gracefully if controller is destroyed", function (assert) {
-    let resolveSuggestions;
+  test("assigning a user via suggestions makes API call and closes the modal", function (assert) {
     pretender.get("/assign/suggestions", () => {
-      return new Promise((resolve) => {
-        resolveSuggestions = resolve;
-      });
+      return [
+        200,
+        { "Content-Type": "application/json" },
+        {
+          suggestions: [],
+          assign_allowed_on_groups: ["nat"],
+          assign_allowed_for_groups: [],
+        },
+      ];
     });
+    pretender.put("/assign/assign", () => {
+      return [200, { "Content-Type": "application/json" }, {}];
+    });
+
+    let modalClosed = false;
     const controller = this.getController("assign-user", {
       model: {
         target: EmberObject.create({}),
       },
+      allowedGroupsForAssignment: ["nat"],
+      taskActions: { allowedGroups: [] },
+    });
+    controller.set("actions.closeModal", () => {
+      modalClosed = true;
     });
 
-    controller.destroy();
-    resolveSuggestions([
-      200,
-      { "Content-Type": "application/json" },
-      {
-        suggestions: [],
-        assign_allowed_on_groups: ["nat"],
-        assign_allowed_for_groups: [],
-      },
-    ]);
```

Comment:
Turns out I have to delete this `isDestroying || isDestroyed` test because there's no good way of simulating a destroyed controller. Ember does not allow me to `set("isDestroyed")` nor recreate the controller after I destroy it.

The only reason it was passing previously was due to the seed which made this run after the subsequent test. With other seeds the controller gets destroyed first then every other test will fail. 😓